### PR TITLE
Always close MongoDB cursor

### PR DESCRIPTION
**Pull request by leleuj**

### CODE REVIEW

File: pac4j-mongo/src/main/java/org/pac4j/mongo/credentials/authenticator/MongoAuthenticator.java [hunk 1]
```diff
@@ -92,10 +92,14 @@ public void validate(UsernamePasswordCredentials credentials) {
...
-            users.add(cursor.next());
-            i++;
+        try {
+            int i= 0;
+            while (cursor.hasNext() && i <= 2) {
+                users.add(cursor.next());
+                i++;
+            }
+        } finally {
+            cursor.close();
```
Comment (mmoayyed) [inline pac4j-mongo/src/main/java/org/pac4j/mongo/credentials/authenticator/MongoAuthenticator.java]: Can this be written using the try-resource syntax?

**Comment by leleuj** [comment]
Good suggestion! Using a "try resource"...

**Comment by leleuj** [comment]
Assuming we're in line, I'll merge this PR tomorrow.

**Comment by mmoayyed** [comment]
Yup. Perfect

**Comment by leleuj** [comment]
Reopened for merge.

**Pull request merged**